### PR TITLE
chore(reuse): ensure ReUSE compliance for Copyright and Licensing

### PR DESCRIPTION
**Pull request by KevinDeJong-TomTom**
Introduces [ReUSE](https://reuse.software) compliance validation to ensure full coverage of Copyright- and License- information. This ensures that:

- All files in the repository are licensed
- Original license texts are enforced in the repository
- [Nice to Have] ability to generate an SPDX 2.3 Software Bill of Materials containing copyright/license information.

In addition, use `dev-build-deploy/reuse-me` to validate compliance as part of the delivery chain.

**NOTE:** This PR is only to open up a discussion on the usefulness of such solution.

### CODE REVIEW

File: .reuse/dep5
Comment (KevinDeJong-TomTom) [inline .reuse/dep5:1]: Please refer to https://reuse.software/spec/#dep5 for more details

**Comment by KevinDeJong-TomTom** [comment]
Generated SPDX SBoM; is attached to this run (https://github.com/tomtom-international/commisery-action/suites/15221328945/artifacts/869850347)

**Comment by FelixFaassen-TomTom** [comment]
@KevinDeJong-TomTom great you are trying out reuse. What exactly do you want me to review in this PR? The most important thing to me in an SBOM are the licenses and copyright info and that is what's missing in the SPDX. Let me know how I can help.

**Review by muggenhor — APPROVED** [review]
Please do the code changes in another PR. Other than that this is fine

**Comment by KevinDeJong-TomTom** [comment]
> @KevinDeJong-TomTom great you are trying out reuse. What exactly do you want me to review in this PR? 

The intention is to see whether there is value in applying enforcement for License/Copyright management in repositories similar to this one, looking at a few perspectives

- Practical additional overhead for a typical engineer
- SCA/SAST; is complementing other initiatives already in place?
- Integration with other tools (i.e. uploading the SPDX SBOM to GitHub Advanced Security)

> The most important thing to me in an SBOM are the licenses and copyright info and that is what's missing in the SPDX. Let me know how I can help.

Regarding missing license information in the SBOM, are you referring to `licenseConcluded`?
Or anything else altogether?

```
    {
      "checksums": [
        {
          "algorithm": "SHA1",
          "checksumValue": "babad0c0157ea4586a39d66322d99427266e7dbe"
        }
      ],
      "fileContributors": [],
      "fileTypes": [],
      "licenseConcluded": "NOASSERTION",
      "licenseInfoInFiles": [
        "CC0-1.0"
      ],
      "attributionTexts": [],
      "fileName": "./.eslintignore",
      "SPDXID": "SPDXRef-88939cd7e826eac4f61dd9adac091b9d64277287",
      "copyrightText": "2023 TomTom <http://tomtom.com>"
    },
```

File: .eslintignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# SPDX-FileCopyrightText: 2023 TomTom <http://tomtom.com>
```
Comment (maikelvdh) [inline .eslintignore:1]: Is it expected to have `http` instead of `https://www.tomtom.com/`?

Comment (Kevin-de-Jong) [reply .eslintignore:1]: I did not dare to change it - however, I’d argue that n URL is not really a contact address and Id expect an email address to TomToms Open Source (or legal) team?

**Comment by Kevin-de-Jong** [comment]
> Please do the code changes in another PR. Other than that this is fine

Fully agreed - mistake on my end

**Pull request closed without merging**